week2/starter/ColorPicker/Views/ContentView.swift

```diff
@@ -0,0 +1,148 @@
+  @Environment(\.horizontalSizeClass) var horizontalSizeClass
+
+  var body: some View {
+    let isBigIphoneLandscape = verticalSizeClass == .compact && horizontalSizeClass == .regular
+    let isSmallIphoneLandscape = verticalSizeClass == .compact && horizontalSizeClass == .compact
+
+    if isBigIphoneLandscape || isSmallIphoneLandscape {
+      LandscapeView(colorPicker: $colorPicker)
+    } else {
+      PortraitView(colorPicker: $colorPicker)
+    }
+  }
+}
+
+struct PortraitView: View {
+  @Binding var colorPicker: ColorPicker
+  @Environment(\.colorScheme) var colorScheme
+
+  var body: some View {
+    VStack {
+      DisplayColorView(colorPicker: $colorPicker)
+        .padding(.bottom)
+      ChangeColorView(colorPicker: $colorPicker)
+    }
+    .background(colorScheme == .dark ? .black : .white)
+    .padding(Constants.General.portraitViewPadding)
+  }
+}
+
+struct LandscapeView: View {
+  @Binding var colorPicker: ColorPicker
+  @Environment(\.colorScheme) var colorScheme
+
+  var body: some View {
+    HStack {
+      DisplayColorView(colorPicker: $colorPicker)
+        .padding(.trailing)
+      ChangeColorView(colorPicker: $colorPicker)
+    }
+    .background(colorScheme == .dark ? .black : .white)
+    .padding(Constants.General.landscapeViewPadding)
+  }
+}
+
+struct DisplayColorView: View {
+  @Binding var colorPicker: ColorPicker
+  @Environment(\.colorScheme) var colorScheme
+
+  var body: some View {
+    VStack {
+      Text("Color Picker")
+        .font(.largeTitle)
+        .bold()
+        .foregroundColor(colorScheme == .dark ? .white : .black)
+      RoundedRectangle(cornerRadius: 0)
+        .foregroundColor(colorPicker.chosenColor)
+        .border(Constants.Colors.displayColorViewBorderColor, width: Constants.General.displayColorViewBorderWidth)
+    }
+  }
+}
+
+struct ChangeColorView: View {
+  @Binding var colorPicker: ColorPicker
+  @Environment(\.colorScheme) var colorScheme
+
+  var body: some View {
+    VStack {
+      ColorSliderView(color: $colorPicker.redSliderValue, name: "Red", accentColor: Constants.Colors.redSliderAccentColor)
+      ColorSliderView(color: $colorPicker.greenSliderValue, name: "Green", accentColor: Constants.Colors.greenSliderAccentColor)
+      ColorSliderView(color: $colorPicker.blueSliderValue, name: "Blue", accentColor: Constants.Colors.blueSliderAccentColor)
+      SetColorButton(colorPicker: $colorPicker)
+    }
+  }
+}
+
+struct SetColorButton: View {
+  @Binding var colorPicker: ColorPicker
+
+  var body: some View {
+    Button {
+      colorPicker.setChosenColor()
+    } label: {
+      Text("Set Color")
+        .padding(Constants.General.buttonPadding)
+        .background(.blue)
+        .foregroundColor(.white)
+        .cornerRadius(Constants.General.buttonCornerRadius)
+        .overlay {
+          RoundedRectangle(cornerRadius: Constants.General.buttonCornerRadius)
+            .strokeBorder(Color.white, lineWidth: Constants.General.buttonStrokeWidth)
+        }
```

Comment:
.buttonStyle(.borderedProminent)
.buttonBorderShape(.roundedRectangle(radius: 21))

But you'd have to make some modifications.